lib/keyrod/fedcloud_client.rb

```diff
@@ -0,0 +1,122 @@
+require 'faraday'
+require 'json'
+require 'net/http'
+
+module Keyrod
+  class FedcloudClient
+    attr_reader :site, :access_token, :auth_path, :ssl
+
+    PROJECTS_PATH = '/v3/auth/projects'.freeze
+    SCOPED_PATH = '/v3/auth/tokens'.freeze
+    REDIRECT_HEADER = 'WWW-Authenticate'.freeze
+    REDIRECT_REGEXP = /(?<=\')(.*?)(?=\')/
+
+    def initialize
+      @site = Keyrod::Settings[:site]
+      @access_token = Keyrod::Settings[:'access-token']
+      @auth_path = "/v3/OS-FEDERATION/identity_providers/#{Keyrod::Settings[:'identity-provider']}/protocols/oidc/auth"
+      @ssl = Keyrod::Settings[:ssl]
+    end
+
+    def unscoped_token
```

Comment:
Methods `unscoped_token`, `projects` and `scoped_token` still looks like the same slightly modified method. Don't tell me you don't see it.